vms/proposervm/batched_vm_test.go

```diff
@@ -583,6 +590,105 @@ func TestBatchedParseBlockPreForkOnly(t *testing.T) {
 	require.Equal(builtBlk3.ID(), res[2].ID())
 }
 
+func TestBatchedParseBlockParallel(t *testing.T) {
+	req := require.New(t)
+
+	parentID := ids.ID{1}
+	timestamp := time.Unix(123, 0)
+	pChainHeight := uint64(2)
+	chainID := ids.GenerateTestID()
+
+	vm := VM{
+		ctx: &snow.Context{ChainID: chainID},
+		ChainVM: &block.TestVM{
+			ParseBlockF: func(_ context.Context, rawBlock []byte) (snowman.Block, error) {
+				return &snowmantest.Block{BytesV: rawBlock}, nil
+			},
+		},
+	}
+
+	tlsCert, err := staking.NewTLSCert()
+	req.NoError(err)
+
+	cert, err := staking.ParseCertificate(tlsCert.Leaf.Raw)
+	req.NoError(err)
+	key := tlsCert.PrivateKey.(crypto.Signer)
+
+	blockThatCantBeParsed := snowmantest.BuildChild(snowmantest.Genesis)
+
+	blockBatch1 := makeParseAbleBlocks(parentID, timestamp, pChainHeight, cert, chainID, key, req)
+	blockBatch1[50] = blockThatCantBeParsed.Bytes()
+
+	blockBatch2 := makeParseAbleBlocks(parentID, timestamp, pChainHeight, cert, chainID, key, req)
+
+	for _, testCase := range []struct {
+		name         string
+		preForkIndex int
+		rawBlocks    [][]byte
+	}{
+		{
+			name: "empty input",
+		},
+		{
+			name:         "pre-fork is somewhere in the middle",
+			rawBlocks:    blockBatch1,
+			preForkIndex: 50,
+		},
+		{
+			name:         "all blocks are post fork",
+			rawBlocks:    blockBatch2,
+			preForkIndex: 9999,
+		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			require := require.New(t)
+			blocks, err := vm.BatchedParseBlock(context.Background(), testCase.rawBlocks)
+			require.NoError(err)
+
+			var expectedBlocks [][]byte
+			for _, block := range blocks {
+				expectedBlocks = append(expectedBlocks, block.Bytes())
+			}
+
+			require.Equal(expectedBlocks, testCase.rawBlocks)
+
+			for i, block := range blocks {
+				if i < testCase.preForkIndex {
+					require.Equal(reflect.TypeOf(block), reflect.TypeOf(&postForkBlock{}))
+				} else {
+					require.Equal(reflect.TypeOf(block), reflect.TypeOf(&preForkBlock{}))
+				}
+			}
+		})
+	}
+}
+
+func makeParseAbleBlocks(parentID ids.ID, timestamp time.Time, pChainHeight uint64, cert *staking.Certificate, chainID ids.ID, key crypto.Signer, req *require.Assertions) [][]byte {
```

Comment:
is it `parseAble` or `parsable`? I think `parsable` is a word right?

Seems like either `parsable` or `parseable` would work: https://en.wiktionary.org/wiki/parsable#English

Reply:
Fixed, thanks!